### PR TITLE
Remove position wrapper methods from Pool

### PR DESCRIPTION
**Pull request by jordanschalm**
This PR consolidates position health getters/setters and sink/source methods by having `Position` access `InternalPosition` directly via a new `EPosition`-scoped `borrowPosition` method, removing 8 thin pass-through methods from `Pool`. Previously there were 3 layers of functions for many position-related operations, that all did the same thing: this PR removes the middle layer where possible.

This PR also adds constraints to `Position.minHealth` to enforce that it is always >1. This has the side effect of fixing an edge case where a user could withdraw from an unhealthy position by manipulating `minHealth` ([this branch contains a test demonstrating the scenario](https://github.com/onflow/FlowCreditMarket/blob/jord/health-constraint-tests/cadence/tests/position_health_constraints_test.cdc#L81-L119)).

- Move all validation into InternalPosition setters (setMinHealth, setMaxHealth, setTargetHealth) so it lives in one place at the lowest level
- Add `EPosition`-scoped lock/unlock to `Pool` for use by `Position` resources
- Rewrite Position health getters/setters and provideSink/provideSource to use `borrowPosition` directly

### CODE REVIEW

File: cadence/contracts/FlowALPv1.cdc [hunk 1]
```diff
@@ -528,19 +528,30 @@ access(all) contract FlowALPv1 {
...
+                minHealth >= 1.0: "Min health (\(minHealth)) must be >1"
+                minHealth <= self.targetHealth: "Min health (\(minHealth)) must be ≤ target health (\(self.targetHealth))"
+            }
+            self.minHealth = minHealth
         }
 
-        /// Sets the Position's maximum health
-        access(EImplementation) fun setMaxHealth(_ value: UFix128) {
-            self.maxHealth = value
+        /// Sets the Position's maximum health. See InternalPosition.minHealth for details.
```
Comment (jordanschalm) [inline cadence/contracts/FlowALPv1.cdc]:
```suggestion
        /// Sets the Position's maximum health. See InternalPosition.maxHealth for details.
```

**Review by zhangchiqing — APPROVED** [review]
LGTM. Also agree to tim's comments.

**Review by Kay-Zee — APPROVED** [review]
LGTM.
